### PR TITLE
Fix generate_addon.sh

### PR DESCRIPTION
**Pull request by futuretea**
Signed-off-by: futuretea <Hang.Yu@suse.com>

1. Fix bug
2. show cloud-init user data，users can copy and paste the content of `cloud-init user data` to **Machine Pools >Show Advanced > User Data**.

**Related issues**
https://github.com/harvester/harvester/issues/2998

### CODE REVIEW

**Review by FrankYang0529 — APPROVED** [review]
LGTM. Test and it works fine.

**Review by chrisho — APPROVED** [review]
Verified on 3 nodes of k3s cluster with cloud-provider.
image
image
It works fine.

LGTM, thanks.

File: deploy/generate_addon.sh [hunk 1]
```diff
@@ -41,14 +41,14 @@ rules:
     resources: [ 'loadbalancers' ]
     verbs: [ 'get', 'watch', 'list', 'update', 'create', 'delete' ]
   - apiGroups: [ 'kubevirt.io' ]
-    resources: [ 'virtualmachines' ]
+    resources: [ 'virtualmachines', 'virtualmachineinstances' ]
```
Comment (guangbochen) [inline deploy/generate_addon.sh:44]: in [here](https://github.com/harvester/harvester/blob/master/deploy/charts/harvester/templates/rbac.yaml#L129-L131) we have requested more verb permission, @yaocw2020 can u please help to confirm whether the `listWatch` permission is also needed?

Comment (futuretea) [reply deploy/generate_addon.sh:44]: watch: https://github.com/harvester/cloud-provider-harvester/blob/ce82c01b7ef617b64481d6e4da0079799eeafb5a/pkg/cloud-controller-manager/loadBalancer.go#L97
no List usage

**Review by guangbochen — APPROVED** [review]
LGTM， thanks

**Pull request merged**